src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] array;
+    private int size = 0;
+
+    public ArrayList() {
+        this.array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == array.length) {
+            capacity();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
 
+        if (size == array.length) {
+            capacity();
+        }
+        if (index == size) {
+            array[size] = value;
+        } else if (index >= 0 && index < size) {
+            System.arraycopy(array, index, array, index + 1, size - index);
+            array[index] = value;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException(index + " is out of bounds.");
```

Comment:
Ensure that the `ArrayListIndexOutOfBoundsException` is properly defined in your project. If it's not defined, you need to create this custom exception class.